src/nagini_translation/lib/resolver.py

```diff
@@ -455,10 +479,26 @@ def _get_subscript_type(node: ast.Subscript, module: PythonModule,
         return module.global_module.classes[INT_TYPE]
     elif value_type.name == SEQ_TYPE:
         return value_type.type_args[0]
+    elif value_type.name == PSET_TYPE:
+        return value_type.type_args[0]
+    elif value_type.python_class.get_func_or_method('__getitem__'):
+        return value_type.python_class.get_func_or_method('__getitem__').type
     else:
         raise UnsupportedException(node)
 
 
+def _get_iteration_type(value_type: PythonType, module: PythonModule,
+                        node: ast.AST) -> PythonType:
+    # Assuming that value_type is some sort of container which supports iteration,
+    # returns the type of the iterator.
+    if value_type.name in (DICT_TYPE, 'defaultdict', 'ExpiringDict'):
+        # FIXME: This is very unfortunate, but right now we cannot handle this
```

Comment:
Ok, but we better not lose track of this. Do we have any place in Nagini's github to add as a something to be done?

Reply:
There is already a very similar FIXME in a different place in the resolver for the same issue, so this isn't new. But you're right, I created an issue about this (number 108).